### PR TITLE
Lang model TCK

### PR DESCRIPTION
**Pull request by dstepanov**
Lang model TCK should pass after there is a new option to skip type annotations checks.
PR implements a custom Annotation model based on the annotation processor which is similar to reflection and TCK requirements. We might want to support both implementations Micronaut/Reflection with some kind of switch.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,11 +3,13 @@ subprojects {
 
     repositories {
         mavenCentral()
+        mavenLocal()
```
Comment (graemerocher) [inline build.gradle]: Remove this

File: processor-cdi/src/main/java/com/oracle/odi/cdi/processor/extensions/AnnotationInfoImpl.java [hunk 2]
```diff
@@ -15,38 +15,53 @@
...
 import io.micronaut.inject.visitor.VisitorContext;
+import jakarta.enterprise.inject.build.compatible.spi.Types;
 import jakarta.enterprise.lang.model.AnnotationInfo;
 import jakarta.enterprise.lang.model.AnnotationMember;
 import jakarta.enterprise.lang.model.declarations.ClassInfo;
 
+import java.util.Map;
+import java.util.stream.Collectors;
+
 class AnnotationInfoImpl implements AnnotationInfo {
```
Comment (graemerocher) [inline processor-cdi/src/main/java/com/oracle/odi/cdi/processor/extensions/AnnotationInfoImpl.java:32]: Add `@Internal`

**Comment by dstepanov** [comment]
The build is based on https://github.com/eclipse-ee4j/cdi-tck/pull/352

File: processor-cdi/src/main/java/com/oracle/odi/cdi/processor/extensions/AnnotationInfoImpl.java [hunk 3]
```diff
@@ -56,21 +71,22 @@ public String name() {
 
     @Override
     public boolean hasMember(String name) {
-        return annotationValue.contains(name);
+        return annotationValue.get(name, Object.class).isPresent();
```
Comment (graemerocher) [inline processor-cdi/src/main/java/com/oracle/odi/cdi/processor/extensions/AnnotationInfoImpl.java:76]: Why was this necessary?

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: processor-cdi/src/main/java/com/oracle/odi/cdi/processor/extensions/AnnotationInfoImpl.java (hunk 3 above)
Comment (dstepanov) [reply processor-cdi/src/main/java/com/oracle/odi/cdi/processor/extensions/AnnotationInfoImpl.java:76]: This actually returns a different value for the repeatable annotation

**Pull request merged**